tools/SetupFlow/DevHome.SetupFlow/Models/RepositoryProvider.cs

```diff
@@ -243,32 +265,84 @@ public IEnumerable<IDeveloperId> GetAllLoggedInAccounts()
     /// </summary>
     /// <param name="developerId">The account to search in.</param>
     /// <returns>A collection of repositories.  May be empty</returns>
-    public IEnumerable<IRepository> GetAllRepositories(IDeveloperId developerId)
+    public IEnumerable<IRepository> GetAllRepositories(IDeveloperId developerId, Dictionary<string, string> searchInputs)
     {
-        IEnumerable<IRepository> repositoriesForAccount;
+        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("CallingExtension", _repositoryProvider.DisplayName, developerId));
 
-        lock (_getRepoLock)
+        RepositoriesResult result;
+        try
         {
-            if (!_repositories.TryGetValue(developerId, out repositoriesForAccount))
+            if (IsSearchingEnabled())
             {
-                TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("CallingExtension", _repositoryProvider.DisplayName, developerId));
-                var result = _repositoryProvider.GetRepositoriesAsync(developerId).AsTask().Result;
-                if (result.Result.Status != ProviderOperationStatus.Success)
-                {
-                    _repositories.Add(developerId, new List<IRepository>());
-                }
-                else
-                {
-                    _repositories.Add(developerId, result.Repositories);
-                }
+                _repositoryProvider2 = _repositoryProvider as IRepositoryProvider2;
+                result = _repositoryProvider2.GetRepositoriesAsync(searchInputs, developerId).AsTask().Result;
+            }
+            else
+            {
+                result = _repositoryProvider.GetRepositoriesAsync(developerId).AsTask().Result;
             }
         }
+        catch (AggregateException aggregateException)
+        {
+            // Because tasks can be cancled DevHome should emit different logs.
+            if (aggregateException.InnerException is OperationCanceledException)
+            {
+                GlobalLog.Logger?.ReportInfo($"Get Repos operation was cancalled.");
+            }
+            else
+            {
+                GlobalLog.Logger?.ReportInfo($"{aggregateException.ToString()}");

```

Comment:
Maybe not since I don't know what this exception would be saying, but should this be an Error? (Not that there's much of a functional difference, but easier to see in logs)